app/controllers/api/v1/digital_objects/uploads_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module DigitalObjects
+      # Because Rails 5 ActiveStorage does not export the BlobRecord class,
+      # we must create the direct upload via REST. In Rails 6, the create logic
+      # below could be moved into a GraphQL mutation, and the returned metadata used
+      # to initiate the ActiveStorage upload.
+      class UploadsController < ApplicationApiController
```

Comment:
Uploading files right now is scoped to digital objects, when we convert this to be a graphql mutation will the upload action be applicable to the upload of all files? One important piece that this current request takes care of is checking permissions, which is checked a second time when the asset is created. Are the two checks necessary? If so, will there be multiple input mutations for each type of upload?

Reply:
Yeah, so this is a big question: We could have an abstracted file upload mutation that just checks valid user, and then clean up the direct upload if the action fails; or we could have two actions. Since I can't implement the independent file upload mutation until Rails 6, I felt like answering that question was out of scope. In the deleted stub code I actually had that very question in the comments.

Reply:
ha, okay, a discussion that can happen later